src/lib/gravityErrorHandler.ts

```diff
@@ -53,50 +57,53 @@ export const formatGravityError = (error) => {
       return {
         type: "error",
         error: freeBody,
-        statusCode: error?.statusCode,
+        statusCode: error.statusCode,
         message: parsedError.error || parsedError.message,
       }
-    } else if (freeBody.error as string) {
+    } else if (freeBody.error) {
       return {
         type: "error",
         message: freeBody.error,
+        statusCode: error.statusCode,
       }
-    } else if (freeBody as string) {
+    } else if (typeof freeBody === "string") {
       return {
         type: "error",
         message: freeBody,
       }
     }
   }
 
-  if (errorSplit && errorSplit.length > 1) {
+  if (error.body) {
     try {
-      const parsedError = JSON.parse(errorSplit[1])
-      const { error, detail, text } = parsedError
-      // check if error message format is an array
-      // see https://github.com/artsy/gravity/blob/master/app/api/util/error_handlers.rb#L32
+      const { error: humanReadableMessage, detail, type } = error.body
+      // check if there is a `detail` array, if so format as an object
+      // and we can return a richer error object
       const fieldErrorResults =
         detail && Object.keys(pickBy(detail, isArray))?.length
 
       if (fieldErrorResults) {
         const fieldErrors = formatGravityErrorDetails(detail)
         return {
           fieldErrors,
-          ...omit(parsedError, "detail"),
+          ...omit(error.body, "detail"),
+          statusCode: error.statusCode,
         }
       }
 
-      if (error) {
+      // if there is a human readable message, return that
+      if (humanReadableMessage) {
```

Comment:
This block _may_ not be needed, and the below which spreads in `...error.body` might be sufficient. I kind of left this sort of as-is, could be removed in the next iteration.